openvpncloud/resource_service.go

```diff
@@ -0,0 +1,307 @@
+package openvpncloud
+
+import (
+	"context"
+	"fmt"
+	"github.com/OpenVPN/terraform-provider-openvpn-cloud/client"
+	"github.com/hashicorp/go-cty/cty"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var (
+	validValues = []string{"ANY", "BGP", "CUSTOM", "DHCP", "DNS", "FTP", "HTTP", "HTTPS", "IMAP", "IMAPS", "NTP", "POP3", "POP3S", "SMTP", "SMTPS", "SNMP", "SSH", "TELNET", "TFTP"}
+)
+
+func resourceService() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceServiceCreate,
+		ReadContext:   resourceServiceRead,
+		DeleteContext: resourceServiceDelete,
+		UpdateContext: resourceServiceUpdate,
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type: schema.TypeString,
+				ValidateDiagFunc: func(i interface{}, path cty.Path) diag.Diagnostics {
```

Comment:
Same here, you can use the `StringLenBetween` validation function https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/validation/strings.go#L71

Reply:
fixed this